modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/FileEntryPicker.tsx

```diff
@@ -0,0 +1,326 @@
+import PropTypes from 'prop-types';
+import React, {ChangeEvent, useEffect, useRef, useState} from 'react';
+
+type TFeedback = {displayType: DisplayType; message: string};
+
+const EMPTY_FEEDBACK: TFeedback = {
+	displayType: 'info',
+	message: '',
+};
+
+const disableFormSubmitButton = (
+	disabled: boolean,
+	portletNamespace: string
+) => {
+	const submitButton = document.getElementById(
+		portletNamespace + 'editClientExtensionEntrySubmitButton'
+	) as HTMLButtonElement;
+
+	if (submitButton) {
+		submitButton.disabled = disabled;
+	}
+};
+
+const readInputFile = (file: File) => {
+	const fileReader = new FileReader();
+
+	return new Promise<string>((resolve) => {
+		fileReader.onload = () => {
+			resolve(fileReader.result as string);
+		};
+
+		fileReader.readAsText(file);
+	});
+};
+
+const FileEntryPicker = ({
+	externalReferenceCode,
+	frontendTokenDefinition: currentFrontendTokenDefinition,
+	learnResources,
+	portletNamespace,
+}) => {
+	const [feedback, setFeedback] = useState(EMPTY_FEEDBACK);
+	const [alertRole, setAlertRole] = useState<'alert' | null>(null);
+	const [isValidatingJSON, setIsValidatingJSON] = useState(false);
+	const [frontendTokenDefinition, setFrontendTokenDefinition] = useState(
+		currentFrontendTokenDefinition
+	);
+
+	const fileInputRef = useRef<HTMLInputElement>();
+	const selectFileButtonRef = useRef<HTMLButtonElement>();
+
+	const setDangerFeedback = (message: string) => {
+		setFeedback({displayType: 'danger', message});
+	};
+
+	const setInfoFeedback = (message: string) => {
+		setFeedback({displayType: 'info', message});
+	};
+
+	const handlePost = async (body: any, endpoint: string) => {
+		return await fetch(
+			'/o/com-liferay-frontend-token-definition-impl' + endpoint,
+			{
+				body,
+				method: 'POST',
+			}
+		);
+	};
+
+	const clearSelection = () => {
+		setFeedback(EMPTY_FEEDBACK);
+
+		setFrontendTokenDefinition('');
+
+		disableFormSubmitButton(false, portletNamespace);
+
+		selectFileButtonRef.current?.focus();
+	};
+
+	const handleFrontendTokenDefinitionFileValidation = async (file: File) => {
+		handlePost(objectToFormData({file}), '/validate-file')
+			.then(async (response) => {
+				return {
+					data: await response.json(),
+					isResponseOk: response.ok,
+				};
+			})
+			.then(({data, isResponseOk}) => {
+				let displayType: DisplayType = 'success';
+
+				if (!isResponseOk) {
+					displayType = 'danger';
+				}
+
+				setFeedback({
+					displayType,
+					message: data.message,
+				});
+
+				disableFormSubmitButton(!isResponseOk, portletNamespace);
+			})
+			.catch(() => {
+				disableFormSubmitButton(true, portletNamespace);
+
+				setDangerFeedback(
+					Liferay.Language.get(
+						'your-upload-failed-to-complete-please-try-again-or-contact-support-if-the-error-persists'
+					)
+				);
+			});
+	};
+
+	const announceJSONValidation = () => {
+		setIsValidatingJSON(true);
+
+		if (!alertRole) {
+			setAlertRole('alert');
```

Comment:
The ClayAlert role is supposed to be 'status', it's 'alert' because there is an issue with the page structure and the behavior of the input file selector.

**Here is the scenario:** when the file is selected the focus goes to the body and i force it to the Select JSON button. The screen reader does not announce the 'status' change since the document lost focus to open the select file modal (which is something from the OS). 

This is what I observed, it might be slightly different regarding where the focus goes and why the screen reader does not announce the change when the role is 'status'. 

To make the screen reader announce the tokens amount, I changed the role to 'alert' which is high priority and is announced as expected.

**Why the alert role start as null?**

When the user opens a existing client extension to visualize its details, a call for the `/tokens-info` endpoint is made to count the tokens and show the green message. If the role starts as 'alert', the Screen Reader will announce the tokens info as soon the page loads and the request is complete. Which is not desired since the user did not upload anything, it was already there.

Reply:
From what I see now, we also should make the tokens info message tabable so the user can hear it while navigating along the FTD field.

Reply:
@ethib137  Does it make sense or am I missing something that it's making things complicated?

Reply:
It makes sense to me but I'm not sure if I understand all the cases here... I think the alert cases can only appear when there is actually an error, i.e. rendering the component over a conditional, but as it seems here you are using this in the case of information, I also think it's ok, the important thing is to announce it to the SR when necessary.

About announcing for the SR with the tab, it is not necessary because when the SR is used the user can use navigation via the cursor and this normally causes each text to be announced... so this is not necessary here.

Reply:
>the important thing is to announce it to the SR when necessary.

It's being announced when necessary, the idea is just to not announce when the page loads and there is already a file selected by the user when the Client Extension was first created.